src/mo_frontend/typing.ml

```diff
@@ -1514,13 +1652,14 @@ and infer_exp'' env exp : T.typ =
       end
     end;
     let ts1 = match pat.it with TupP _ -> T.seq_of_tup t1 | _ -> [t1] in
+    let sort = if is_flexible && sort = T.Local T.Stable then T.Local T.Flexible else sort in
```

Comment:
So this is implicitly weakening the sort to flexible if necessary.
What happens if the function was recursive, and you assumed it was stable for all recursive calls, but then decided it was flexible after all. Can that lead to problems?

Reply:
No, this is only because the parser (due LR) does not yet consider nested scopes and just gives an initial classification on stable/flexible. Later on, I consider the nested scopes, and if the surrounding scope is flexible, a parser-assumed stable keyword is weakened to flexible too. A LL-parser could handle that. Or, ideally, this classification could be done on symbol table only. And the parser only records explicit `stable` use, no implied annotations yet.